src/pages/resultsView/download/CaseAlterationTable.tsx

```diff
@@ -33,71 +39,105 @@ export interface ICaseAlterationTableProps {
+                getAlterationData: (oqlData: IOqlData) => oqlData.mutation,
+                isNotProfiled: (oqlData: IOqlData) => oqlData.isMutationNotProfiled,
+                getValues: stringMapper
+            }
+            break;
+        case "FUSION":
+            generator = {
+                label: "FUSION",
+                getAlterationData: (oqlData: IOqlData) => oqlData.fusion,
+                isNotProfiled: (oqlData: IOqlData) => oqlData.isFusionNotProfiled,
+                getValues: stringMapper
+            }
+            break;
+        case "CNA":
+            generator = {
+                label: "CNA",
+                getAlterationData: (oqlData: IOqlData) => oqlData.cna,
+                isNotProfiled: (oqlData: IOqlData) => oqlData.isCnaNotProfiled,
+                getValues: subAlterationMapper
+            }
+            break;
+        case "EXP":
+            generator = {
+                label: "EXP",
+                getAlterationData: (oqlData: IOqlData) => oqlData.mrnaExp,
+                isNotProfiled: (oqlData: IOqlData) => oqlData.isMrnaExpNotProfiled,
+                getValues: subAlterationMapper
+            }
+            break;
+        case "PROT":
+            generator = {
+                label: "PROT",
+                getAlterationData: (oqlData: IOqlData) => oqlData.proteinLevel,
+                isNotProfiled: (oqlData: IOqlData) => oqlData.isProteinLevelNotProfiled,
+                getValues: subAlterationMapper
+            }
+            break;
+        default:
+    }
 
+    if (generator) {
+        const alterationData = generator.getAlterationData(data);
         if (alterationData.length > 0) {
-            oqlValue.push(`${generator.label}: ${generator.getValues(alterationData).join(",")};`);
+            pseudoOqlSummary = generator.getValues(alterationData).join(",");
         }
-    });
+        if (generator.isNotProfiled(data)) {
+            pseudoOqlSummary = "not profiled";
+        }
+    }
 
     // finally, generate a single line summary with all alteration data combined.
-    return oqlValue.join(" ");
+    return pseudoOqlSummary;
 }
 
-export function generateOqlDisplayValue(oqlData: {[oqlLine: string]: IOqlData}, oqlLine: string)
+export function generatePseudoOqlSummary(oqlData: {[oqlLine: string]: IOqlData}, oqlLine: string, alterationType: string)
 {
-    let oqlValue = "";
+    let pseudoOqlSummary = "";
 
     if (!_.isEmpty(oqlData))
     {
         const datum = oqlData[oqlLine];
 
         if (datum) {
-            oqlValue = datum.sequenced ? generateOqlValue(oqlData[oqlLine]) : "N/P";
+            pseudoOqlSummary = datum.sequenced ? generateOqlValue(oqlData[oqlLine], alterationType) : "no alteration";
         }
     }
 
-    return oqlValue;
+    return pseudoOqlSummary;
+}
+
+export function computeAlterationTypes(alterationData: ICaseAlteration[]): string[]
+{
+    const types = new Set<string>();
+    alterationData.forEach(element => {
```

Comment:
this code could be clarified with use of lodash.  you're basically trying to merge the values of alterationTypes collection for teach item in oqlData.   you should be using _.map and _.values and also look at _.chain to make things more succinct and easy to ready.